SAML: Allow disabling of SAML signups (#47481)
grafana/grafana: docs/sources/enterprise/saml.md

| ---------------------------------------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
| `enabled`                                                  | No       | Whether SAML authentication is allowed                                                                                                                                                              | `false`       |
| `single_logout`                                            | No       | Whether SAML Single Logout enabled                                                                                                                                                                  | `false`       |

> | `allow_sign_up`                                            | No       | Whether to allow new Grafana users to be created through SAML login. If set to `false`, only pre-existing Grafana users will be able to login with SAML.                                            | `true`        |

Reviewer: ```suggestion
| `allow_sign_up`                                            | No       | Whether to allow new Grafana user creation through SAML login. If set to `false`, then only existing Grafana users can log in with SAML.                                            | `true`        |
```

Grafana provides configuration options that let you modify which keys to look at for these values. The data we need to create the user in Grafana is Name, Login handle, and email.

### Allow new user signups

> By default, any new Grafana user who authenticates with SAML will have an account created for them automatically. To decouple authentication and account creation and ensure only users with existing accounts can log in with SAML, set the `allow_sign_up` option to false.

Reviewer: ```suggestion
By default, new Grafana users using SAML authentication will have an account created for them automatically. To decouple authentication and account creation and ensure only users with existing accounts can log in with SAML, set the `allow_sign_up` option to false.
```
